src/fetcher.js

```diff
@@ -1,129 +1,115 @@
 import AWS from 'aws-sdk'
-import request  from 'request'
-import moment from 'moment'
+import request from 'request'
 import {fetchConfig} from './config'
+import NamedError from './NamedError'
 
-let UPLOAD_BUCKET = 'fulfilment-output-test';
-let CONFIG_BUCKET = 'fulfilment-private';
+let UPLOAD_BUCKET = 'fulfilment-output-test'
 
-//todo see if we can use the same for both things
-let s3 = new AWS.S3();
+let uploadS3 = new AWS.S3()
 
-let uploadS3 = new AWS.S3({signatureVersion: 'v4' });
-
-function fetchFile(batch, deliveryDate, config){
-	let promise = new Promise((resolve, reject) => {
-		console.log(`fetching file from zuora with id ${batch.fileId}`);
-		const options = {                 
-		  method: 'GET',             
-		  uri: `${config.url}/apps/api/batch-query/file/${batch.fileId}`,   
-		  json: true,                  
-		  headers: {               
-		    'Authorization': 'Basic ' + new Buffer(`${config.username}:${config.password}`).toString('base64'),
-		    'Content-Type': 'application/json'                  
-		  }
-		};
-		request(options, function(error, response, body) {  
-		if (error) {
-			reject(getError('api_call_error',error));
-			return;
-			}	
-		  if(response.statusCode != '200') {
-		  		reject(getError('api_call_error', `error response status ${response.statusCode} when getting batch ${batch.name}`));
-			} else {
-				//TODO SEE HOW TO DETECT FAILURES OR ANY OTHER SPECIAL CASE HERE
-				let fileData = {
-					batchName : batch.name,
-					fileName : `${batch.name}_${deliveryDate}.csv`,
-					data : body
-				};	
-				resolve(fileData);
-			}
-		});
-	});
-	return promise;
+function fetchFile (batch, deliveryDate, config) {
+  let promise = new Promise((resolve, reject) => {
+    console.log(`fetching file from zuora with id ${batch.fileId}`)
+    const options = {
+      method: 'GET',
+      uri: `${config.url}/apps/api/batch-query/file/${batch.fileId}`,
+      json: true,
+      headers: {
+        'Authorization': 'Basic ' + Buffer.from(`${config.username}:${config.password}`).toString('base64'),
+        'Content-Type': 'application/json'
+      }
+    }
+    request(options, function (error, response, body) {
+      if (error) {
+        reject(new NamedError('api_call_error', error))
+        return
+      }
+      if (response.statusCode !== '200') {
```

Comment:
i'm not sure if the status is a string or an int here, comparing it like this might not work